File: practice/src/main/java/com/decade/practice/usecases/core/OnlineStatistic.java
package com.decade.practice.usecases.core;

import com.decade.practice.entities.OnlineStatus;
import com.decade.practice.entities.domain.entity.User;

import java.time.Instant;
import java.util.List;

public interface OnlineStatistic {

      OnlineStatus set(User user, long at);

      default OnlineStatus set(User user) {
            return set(user, Instant.now().getEpochSecond());
      }

      OnlineStatus get(String username);

      List<OnlineStatus> getOnlineList(String username);
}

File: practice/src/main/java/com/decade/practice/core/OnlineStatistic.kt
package com.decade.practice.core

import com.decade.practice.model.OnlineStatus
import com.decade.practice.model.domain.entity.User
import java.time.Instant

interface OnlineStatistic {
      fun set(user: User, at: Long = Instant.now().epochSecond): OnlineStatus
      fun get(username: String): OnlineStatus
      fun getOnlineList(username: String): List<OnlineStatus>
}